### PR TITLE
Add nullability annotations to signature package and its dependencies

### PR DESCRIPTION
**Pull request by SUPERCILEX**

### CODE REVIEW

**Comment by sjudd** [comment]
I believe it's a flaky failure, filed https://github.com/bumptech/glide/issues/2754

File: library/src/main/java/com/bumptech/glide/signature/MediaStoreSignature.java [hunk 1]
```diff
@@ -26,8 +27,8 @@
    * @param orientation  The orientation of the media store media. Ok to default to 0. See {@link
    *                     android.provider.MediaStore.Images.ImageColumns#ORIENTATION}.
    */
-  public MediaStoreSignature(String mimeType, long dateModified, int orientation) {
-    this.mimeType = mimeType;
+  public MediaStoreSignature(@Nullable String mimeType, long dateModified, int orientation) {
+    this.mimeType = mimeType == null ? "" : mimeType;
```
Comment (sjudd) [inline library/src/main/java/com/bumptech/glide/signature/MediaStoreSignature.java]: Add @NonNull to the mimeType instance variable?

Comment (SUPERCILEX) [reply library/src/main/java/com/bumptech/glide/signature/MediaStoreSignature.java]: SGTM.

**Comment by sjudd** [comment]
Thanks!

File: library/src/main/java/com/bumptech/glide/signature/ApplicationVersionSignature.java [hunk 2]
```diff
@@ -20,7 +21,8 @@
    * Returns the signature {@link com.bumptech.glide.load.Key} for version code of the Application
    * of the given Context.
    */
-  public static Key obtain(Context context) {
+  @NonNull
+  public static Key obtain(@NonNull Context context) {
     String packageName = context.getPackageName();
     Key result = PACKAGE_NAME_TO_KEY.get(packageName);
```
Comment (anticafe) [inline library/src/main/java/com/bumptech/glide/signature/ApplicationVersionSignature.java]: We can annotate `PACKAGE_NAME_TO_KEY` as @NonNull too

File: library/src/main/java/com/bumptech/glide/signature/EmptySignature.java [hunk 3]
```diff
@@ -9,6 +10,7 @@
 public final class EmptySignature implements Key {
   private static final EmptySignature EMPTY_KEY = new EmptySignature();
```
Comment (anticafe) [inline library/src/main/java/com/bumptech/glide/signature/EmptySignature.java]: Yeah, `EMPTY_KEY` is @NonNull too

Comment (sjudd) [reply library/src/main/java/com/bumptech/glide/signature/EmptySignature.java]: @anticafe thanks for pointing this out, mind sending a pull request to update those too? Thanks!

**Pull request merged**